plugins/discourse-local-dates/assets/javascripts/initializers/discourse-local-dates.js

```diff
@@ -124,18 +124,24 @@ function _rangeElements(element) {
     return [];
   }
 
-  // TODO: element.parentElement.children.length !== 2 is a fallback to old solution for ranges
-  // Condition can be removed after migration to [date-range]
-  if (
-    element.dataset.range !== "true" &&
-    element.parentElement.children.length !== 2
-  ) {
-    return [element];
+  if (element.dataset.range) {
+    return _partitionedRanges(element).find((pair) => pair.includes(element));
   }
 
-  return Array.from(element.parentElement.children).filter(
-    (span) => span.dataset.date
+  return [element];
+}
+
+function _partitionedRanges(element) {
```

Comment:
This function assumes date-range elements always come in pairs of two. I decided not to be defensive about it, as there is nothing we could do to recover at this point if something is wrong upstream.